### PR TITLE
OCM-177: Move all identity providers to internal pkg

### PR DESCRIPTION
**Pull request by sagidayan**
All IDPs are now located in /provider/idps/

this includes:
 - schemas
 - structs (states)
 - validations (if exists)
 - updated golang string slices to terraform types.List
 - Added missing descriptions + updated docs
 - provider builders

### CODE REVIEW

File: provider/idps/ldap.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	BindDN       types.String                    `tfsdk:"bind_dn"`
+	BindPassword types.String                    `tfsdk:"bind_password"`
+	CA           types.String                    `tfsdk:"ca"`
+	Insecure     types.Bool                      `tfsdk:"insecure"`
+	URL          types.String                    `tfsdk:"url"`
+	Attributes   *LDAPIdentityProviderAttributes `tfsdk:"attributes"`
+}
+
+type LDAPIdentityProviderAttributes struct {
+	EMail             []string `tfsdk:"email"`
```
Comment (bardielle) [inline provider/idps/ldap.go:21]: Can you update the type to be types.List?

File: provider/idps/ldap.go [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+	ID                []string `tfsdk:"id"`
+	Name              []string `tfsdk:"name"`
+	PreferredUsername []string `tfsdk:"preferred_username"`
+}
+
+func LdapSchema() tfsdk.NestedAttributes {
+	return tfsdk.SingleNestedAttributes(map[string]tfsdk.Attribute{
```
Comment (bardielle) [inline provider/idps/ldap.go:28]: I know you just copied it 
but the description is missing here...

**Review by bardielle — COMMENTED** [review]
Nice job! 
I had some comments

**Comment by sagidayan** [comment]
/hold

Will update according to @bardielle comments

File: provider/idps/ldap.go (hunk 1 above)
Comment (sagidayan) [reply provider/idps/ldap.go]: Done

File: provider/idps/ldap.go (hunk 2 above)
Comment (sagidayan) [reply provider/idps/ldap.go:29]: Done

File: provider/idps/ldap.go (hunk 1 above)
Comment (bardielle) [reply provider/idps/ldap.go]: I think you also need to change the relevant code that set those values

**Review by bardielle — APPROVED** [review]
/lgtm

**Pull request merged**